core/src/main/scala/kafka/log/TimeIndex.scala

```diff
@@ -256,8 +265,49 @@ class LazyTimeIndex(@volatile private var _file: File, baseOffset: Long, maxInde
   }
 
   def get: TimeIndex = {
+    if (isClosed)
+      throw new IllegalStateException(s"Attempt to access the closed TimeIndex (file=${_file}, baseOffset=${baseOffset}, " +
+                                      s"maxIndexSize=${maxIndexSize}, writable=${writable}.")
     if (timeIndex.isEmpty)
       timeIndex = Some(new TimeIndex(_file, baseOffset, maxIndexSize, writable))
     timeIndex.get
   }
+
+  /**
+   * Close this index file.
+   * Note: This will be a no-op if the index has already been closed.
+   */
+  def close(): Unit = {
+    if (!isClosed) {
+      timeIndex.foreach(_.close())
```

Comment:
use timeIndex.isDefined?   foreach raises possible confusion of multiple index files.

Reply:
`<Option>.foreach(_.close())` seems to be a commonly used Scala pattern used across Kafka code base  -- e.g. please see 

1. [TransactionIndex#close()](https://github.com/linkedin/kafka/blob/2.3-li/core/src/main/scala/kafka/log/TransactionIndex.scala#L100),
2. [QuotaFactory#shutdown()](https://github.com/linkedin/kafka/blob/2.3-li/core/src/main/scala/kafka/server/QuotaFactory.scala#L52),
3. [AdminManager#shutdown()](https://github.com/linkedin/kafka/blob/2.3-li/core/src/main/scala/kafka/server/AdminManager.scala#L531), or
4. [KafkaServer#shutdown()](https://github.com/linkedin/kafka/blob/2.3-li/core/src/main/scala/kafka/server/KafkaServer.scala#L646)

Also the JavaDoc indicates that this class is `A thin wrapper on top of the raw TimeIndex object` -- i.e. not a collection of `TimeIndex` objects.